gateways/Gateways.md

```diff
@@ -13,18 +13,24 @@ All fjåge Gateway implementations should implement the following classes and me
 - Must return a `null` if the connection to master container fails on the first attempt.
 - May support auto-reconnect, where, once a connection with the master container is established if the connection fails, the Gateway tries to reconnect automatically.
 
-### `getAgentID()` :: Void -> AgentID
+### `authenticate` :: String credentials -> Boolean
+- Tries to authenticate the Gateway connection with the master container.
+- Must send the credential string using command `{"action":"auth","creds":"boo"}`
+- Must wait for a the first `{"auth": true}` or `{"auth": false}` received from the master and return the result.
+- Must default timeout to 1000 millisecond.
+
+### `getAgentID()` :: Void -> AgentID | Error
 -  Returns the _AgentID_ associated with the gateway.
 -  May be implemented as a property `agentID` on the _Gateway_.
 
 ### `close()` :: Void -> Void
 - Closes the _Gateway_.
 - Must send a `{"alive": false}` message to the master container before closing.
 
-### `send()` :: Message -> Boolean
+### `send()` :: Message -> Boolean  
 - Sends a _Message_ to the recipient indicated in the message.
 
-### `receive()` :: (Object filter), (Int timeout) -> Message
+### `receive()` :: (Object filter), (Int timeout) -> Message | Error
```

Comment:
returning `Error` makes sense for some languages, other places it is throwing error. Do we need to define it as a return value?

Reply:
As discussed in #167, can't do this generally.